### ISSUE TITLE
Race condition freezing class loading causing server hang

### ISSUE THREAD

**Issue by gabizou** [DimensionalDevelopment/JustEnoughIDs#102]
Just stumbling across this from a user report, but it appears that [this line](https://github.com/DimensionalDevelopment/JustEnoughIDs/blob/ccc581a76fefcd605c58897599f68e967a5bb2ec/src/main/java/org/dimdev/jeid/JEIDTransformer.java#L67) is causing a deadlock in early stages of the game's load time.

The provided log: https://gist.github.com/gabizou/c602c34407d3f2641f22a931614d5cfc#file-gistfile1-txt-L12 

points to a Stream operation being executed, and that operation shows up in two individual threads, [here](https://gist.github.com/gabizou/c602c34407d3f2641f22a931614d5cfc#file-gistfile1-txt-L111) and [here](https://gist.github.com/gabizou/c602c34407d3f2641f22a931614d5cfc#file-gistfile1-txt-L127). 

Upon further investigation, I'm questioning **why** someone would decide it's best to try and parallelize method lookups during class transformation... What do you gain out of it?

**Comment by ZombieHDGaming** [comment]
I'll be perfectly honest, Zabi was nice enough to help me with that when it came to the potion extension since I'm not as talented with Mixins yet, so he helped me do it in pure ASM. I'll see about making it so it's not done in parallel and see if that possibly stops it (Idk how to test it, I've never personally seen the issue), or even better, port the potion extensions to pure Mixins finally since maybe Mixins 0.8 might help that more

**Comment by zabi94** [comment]
Yeah, that was a derp on my part, I was experimenting with streams and that bit got left behind. That also explains one old bug I had on my standalone patch

**Comment by clienthax** [comment]
I can confirm making it a standard stream resolves the hang on my setup (I was the one with the stack dump)
